evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -130,3 +141,16 @@ def test_iterative_algorithm_results(ensembling_value, dummy_binary_pipeline_cla
             for score, pipeline in zip(scores, next_batch):
                 algo.add_result(score, pipeline)
             assert pipeline.model_family == ModelFamily.ENSEMBLE
+
+
+def test_iterative_algorithm_instantiates_text():
+    class MockTextClassificationPipeline(BinaryClassificationPipeline):
+        estimator = MockEstimator
+        component_graph = [TextFeaturizer, MockEstimator]
+
+    algo = IterativeAlgorithm(allowed_pipelines=[MockTextClassificationPipeline], text_columns=['text_col_1', 'text_col_2'])
+    pipeline = algo.next_batch()[0]
+    expected_params = {'text_columns': ['text_col_1', 'text_col_2']}
+    assert pipeline.parameters['Text Featurization Component'] == expected_params
+    assert isinstance(pipeline.component_graph[0], TextFeaturizer)
```

Comment:
Hmm, wouldn't `pipeline.component_graph[0]` be the imputer in this case, and you want `pipeline.component_graph[1]`? What am I missing?

Reply:
The pipeline in this test isn't dynamically generated, it's defined a few lines above as only having the TextFeaturizer and an estimator

Reply:
Ah right, thanks!